chore(ci): build Falco in RelWithDebInfo, and upload Falco debug symbols as github artifacts

FedeDP opened the pull request:
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area CI

**What this PR does / why we need it**:

https://github.com/falcosecurity/falco/pull/3440 allows us to build Falco in RelWithDebInfo target.
Use the new target and upload the debug symbols as github artifacts.
Then, as part of the release CI, use these artifacts and upload them as github assets.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
chore(ci): build Falco in RelWithDebInfo, and upload Falco debug symbols as github artifacts
```

FedeDP commented:
/milestone 0.40.0

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/3452#" title="Author self-approved">FedeDP</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/falco/blob/master/OWNERS)~~ [FedeDP]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

FedeDP commented:
Ok PR tested:
* downloaded x86_64 tar.gz (release build) from the workflow artifacts
* dowloaded x86_64 debug symbols from the release artifacts
* extracted both
* moved `falco.debug` near to `falco` executable
* run `gdb ./falco` and you will be prompted with:
> Reading symbols from ./falco...
> Reading symbols from /home/federico/Scaricati/falco-0.40.0-113+3ccd084-x86_64/usr/bin/falco.debug...


Also:
```
file falco
falco: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.0.0, BuildID[xxHash]=d425e7ff0ec684c0, stripped

file falco.debug 
falco.debug: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter *empty*, for GNU/Linux 2.0.0, BuildID[xxHash]=d425e7ff0ec684c0, with debug_info, not stripped
```

leogr reviewed (approved):
whoa!

FedeDP commented:
Also, thanks to @shane-lawrence for the `RelWithDebInfo` cmake target support!!

LucaGuerra commented on .github/workflows/release.yaml:
```diff
@@ -160,9 +160,28 @@ jobs:
           tag_name: ${{ github.event.release.tag_name }}
           name: ${{ github.event.release.name }}
+          files: |
+            falco-x86_64.debug
+            falco-aarch64.debug
```
Did `_` at line 169 and 177 become `-`?

FedeDP replied:
Oh thanks!

FedeDP replied:
Fixed!
